File: feign-persistence/src/main/java/name/abhijitsarkar/feign/persistence/service/MongoDbRecordingService.java
/*
 * Copyright (c) 2016, the original author or authors.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * License for more details.
 */

package name.abhijitsarkar.feign.persistence.service;

import lombok.extern.slf4j.Slf4j;
import name.abhijitsarkar.feign.persistence.RecordingRequest;
import name.abhijitsarkar.feign.persistence.RecordingService;
import name.abhijitsarkar.feign.persistence.repository.MongoDbRequestRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.PayloadApplicationEvent;
import org.springframework.context.event.EventListener;

/**
 * @author Abhijit Sarkar
 */
@Slf4j
@SuppressWarnings({"PMD.BeanMembersShouldSerialize"})
public class MongoDbRecordingService implements RecordingService {
    @Autowired
    MongoDbRequestRepository mongoDbRequestRepository;

    @EventListener
    public void record(PayloadApplicationEvent<RecordingRequest> requestEvent) {
        RecordingRequest recordingRequest = requestEvent.getPayload();
        String id = recordingRequest.getId();

        log.info("Recording request with id: {}.", id);

        mongoDbRequestRepository.save(recordingRequest);
    }
}

File: feign-persistence/src/main/scala/name/abhijitsarkar/feign/persistence/service/MongoDbRecordingService.scala
/*
 * Copyright (c) 2016, the original author or authors.
 *
 * This program is free software; you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation; version 3 of the License.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * A copy of the GNU General Public License accompanies this software,
 * and is also available at http://www.gnu.org/licenses.
 */

package name.abhijitsarkar.feign.persistence.service

import name.abhijitsarkar.feign.persistence.repository.MongoDbRequestRepository
import name.abhijitsarkar.feign.persistence.{RecordingRequest, RecordingService}
import org.slf4j.LoggerFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.context.PayloadApplicationEvent
import org.springframework.context.event.EventListener

/**
  * @author Abhijit Sarkar
  */
class MongoDbRecordingService @Autowired()(private val mongoDbRequestRepository: MongoDbRequestRepository)
  extends RecordingService {
  val logger = LoggerFactory.getLogger(classOf[MongoDbRecordingService])

  @EventListener
  def record(requestEvent: PayloadApplicationEvent[RecordingRequest]) {
    val recordingRequest: RecordingRequest = requestEvent.getPayload
    val id: String = recordingRequest.getId

    logger.info("Recording request with id: {}.", id)

    mongoDbRequestRepository.save(recordingRequest)
  }
}
